source

```diff
@@ -95411,6 +95411,38 @@ location.href = '#foo';</code></pre>
         <li><p>Set <var>document</var>'s <i data-x="concept-document-salvageable">salvageable</i> state to false.</p></li>
       </ol>
 
+  <p>To <dfn>build NotRestoredReasons object for document</dfn> given <code>Document</code>
+  <var>childDocument</var> and <code>Document</code> <var>rootDocument</var>:</p>
```

Comment:
It looks like `rootDocument` is not used? Then maybe we can remove it, and also not use "child" for `childDocument`, as it can refer to a main document as well right?

Reply:
Sorry my change was not uploaded completely, now it has all the changes. PTAL again

Reply:
Thanks. But I think it might still be better for us to remove it from here to simplify it, by just adding the reasons regardless of origin. Then from [build notRestoredReasons object for page](https://docs.google.com/document/d/1emdzkJW-vUrwYU1I36UPOyO4QL5d15Ie6pbPXJtmWkY/edit?resourcekey=0-Iai2yb3jPaphdVfM2Cvkwg#bookmark=id.a2pxqletlpy9) you can redact the reasons for same-origin descendants (since you are already going through cross-origin descendants there anyways)?

Reply:
Thanks, I removed the rootDocument, PTAL again